src/main/java/gross/gameoflife/Grid.java

```diff
@@ -0,0 +1,224 @@
+package gross.gameoflife;
+
+import java.util.ArrayList;
+
+public class Grid {
+
+    private int[][] board;
+
+    // grid constructor
+    public Grid(int height, int width) {
+        board = new int[height][width];
+    }
+
+    // method to set a living cell
+    public void setCellAlive(int row, int column) {
+        if (row >= board.length || column >= board[row].length) {
+            System.out.println("Action aborted. Provided dimensions are out of bounds.");
+        } else if (row < 0 || column < 0) {
+            System.out.println("Action aborted. Provided dimensions are negative.");
+        } else {
+            board[row][column] = 1;
+        }
+    }
+
+    public void setCellDead(int row, int column) {
+        if (row >= board.length || column >= board[row].length) {
+            System.out.println("Action aborted. Provided dimensions are out of bounds.");
+        } else if (row < 0 || column < 0) {
+            System.out.println("Action aborted. Provided dimensions are negative.");
+        } else {
+            board[row][column] = 0;
+        }
+    }
+
+    // produces next generation of cells according to Conway's rules
+    public void nextGen() {
+        ArrayList<int[]> cellsToDie = new ArrayList<>();
+        ArrayList<int[]> cellsToComeAlive = new ArrayList<>();
+
+        // traverse the entire 2D array to identify all living cells
+        for (int i = 0; i < board.length; i++) {
+            for (int j = 0; j < board[i].length; j++) {
+                int cell = board[i][j];
+                // if cell is alive, check status of neighbors
+                if (cell == 1) {
+                    // identify all live neighbors
+                    int liveNeighbors = findLiveNeighbors(i, j);
+                    // will die if it does not have 2 or 3 neighbors (1 or less, 4 or more)
+                    if (liveNeighbors != 2 && liveNeighbors != 3) {
+                        int[] cellLocation = {i, j};
+                        cellsToDie.add(cellLocation);
+                    }
+                    // identify all dead neighbors
+                    ArrayList<int[]> deadNeighbors = findDeadNeighbors(i, j);
+                    // loop through all dead cells, check if they have 3+ neighbors
+                    for (int[] deadCell : deadNeighbors) {
+                        boolean alreadyUsed = false;
+                        for (int[] c : cellsToComeAlive) {
+                            if (c[0] == deadCell[0] && c[1] == deadCell[1]) {
+                                alreadyUsed = true;
+                            }
+                        }
+                        if (!alreadyUsed) {
+                            // add to cellsToComeAlive if not already there
+                            if (findLiveNeighbors(deadCell[0], deadCell[1]) == 3) {
+                                cellsToComeAlive.add(deadCell);
```

Comment:
A simpler way to do this is to have two int[][] arrays, not one.

Reply:
I'm not sure what you mean. One 2D array, or two regular arrays? I am also not sure which variable you are referring to. I have a lot of arrays and Array Lists here. Which do you think should be replaced?